test/e2e/main_test.go

```diff
@@ -452,3 +453,37 @@ func testServerTLS(ctx context.Context, namespace string) func(t *testing.T) {
 		}
 	}
 }
+
+// TestIsManagedByController test prometheus operator managing object with correct ControlerID.
+func testMultipleOperators(testCtx *operatorFramework.TestCtx) func(t *testing.T) {
+	return func(t *testing.T) {
+		skipPrometheusAllNSTests(t)
+
+		ns := framework.CreateNamespace(context.Background(), t, testCtx)
+		// Create operator-2 in a new ns and set controller-id.
+		finalizers, err := framework.CreateOrUpdatePrometheusOperatorWithOpts(context.Background(),
+			operatorFramework.PrometheusOperatorOpts{
+				Namespace:           ns,
+				ClusterRoleBindings: true,
+				EnableScrapeConfigs: true,
+				AdditionalArgs:      []string{"--controller-id=42"},
```

Comment:
can you use `testControllerID` instead?

Reply:
sure, let's check it

Reply:
Ok, now the issue is, after change: 
`controllerIDAnnotation = "operator.prometheus.io/wrong-label"`
controllerID of the object is empty and controllerID of controller if empty, then object is created :S

How could we fix it?

Reply:
:facepalm: of course, sorry for my wrong suggestion. So I don't thing we need to be paranoid. Please revert that part and we can merge.